tornado/test/httpserver_test.py

```diff
@@ -423,18 +439,22 @@ def test_double_slash(self):
         self.assertEqual(200, response.code)
         self.assertEqual(json_decode(response.body), {})
 
-    def test_malformed_body(self):
-        # parse_qs is pretty forgiving, but it will fail on python 3
-        # if the data is not utf8.
-        with ExpectLog(gen_log, "Invalid x-www-form-urlencoded body"):
-            response = self.fetch(
-                "/echo",
-                method="POST",
-                headers={"Content-Type": "application/x-www-form-urlencoded"},
-                body=b"\xe9",
-            )
-        self.assertEqual(200, response.code)
-        self.assertEqual(b"{}", response.body)
+    def test_form_urlencode(self):
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        uni_text = "chinese: \u5f20\u4e09"
+        for enc in ("utf8", "gbk"):
+            body = b"data=" + urllib.parse.quote(uni_text.encode(enc)).encode()
```

Comment:
Doesn't that last `encode()` need an argument? (a non-ascii-compatible system default encoding is unlikely, but still)

Reply:
The default for `str.encode()` is always utf-8, it is not based on the locale (like e.g. printing unicode strings to stdout is). I could encode as 'ascii' here as a sort of documentation though.
https://docs.python.org/3.5/library/stdtypes.html#str.encode